clients/src/main/java/org/apache/kafka/clients/consumer/RoundRobinAssignor.java

```diff
@@ -0,0 +1,125 @@
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * The roundrobin assignor lays out all the available partitions and all the available consumers. It
+ * then proceeds to do a roundrobin assignment from partition to consumer. If the subscriptions of all consumer
+ * instances are identical, then the partitions will be uniformly distributed. (i.e., the partition ownership counts
+ * will be within a delta of exactly one across all consumers.)
+ *
+ * For example, suppose there are two consumers C0 and C1, two topics t0 and t1, and each topic has 3 partitions,
+ * resulting in partitions t0p0, t0p1, t0p2, t1p0, t1p1, and t1p2.
+ *
+ * The assignment will be:
+ * C0: [t0p0, t0p2, t1p1]
+ * C1: [t0p1, t1p0, t1p2]
+ */
+public class RoundRobinAssignor extends AbstractPartitionAssignor {
+
+    @Override
+    public List<TopicPartition> assign(String consumerId,
+                                       Map<String, Integer> partitionsPerTopic,
+                                       Map<String, ConsumerMetadata> consumers) {
+        Map<String, List<TopicPartition>> assignment = new HashMap<>();
+        CircularIterator<String> assigner = new CircularIterator<>(sorted(consumers.keySet()));
+        for (TopicPartition partition : allPartitionsSorted(partitionsPerTopic, consumers)) {
+            final String topic = partition.topic();
+            while (!consumers.get(assigner.peek()).topics().contains(topic))
+                assigner.next();
+            put(assignment, assigner.next(), partition);
+        }
+
+        List<TopicPartition> assignedPartitions = assignment.get(consumerId);
+        if (assignedPartitions == null)
+            return Collections.emptyList();
+
+        return assignedPartitions;
+    }
+
+    public <T extends Comparable<T>> List<T> sorted(Collection<T> consumers) {
+        List<T> res = new ArrayList<>(consumers);
+        Collections.sort(res);
+        return res;
+    }
+
+    public List<TopicPartition> allPartitionsSorted(Map<String, Integer> partitionsPerTopic,
+                                                    Map<String, ConsumerMetadata> consumerMetadata) {
+        SortedSet<String> topics = new TreeSet<>();
+        for (ConsumerMetadata metadata : consumerMetadata.values())
+            topics.addAll(metadata.topics());
+
+        List<TopicPartition> allPartitions = new ArrayList<>();
+        for (String topic : topics) {
+            Integer partitions = partitionsPerTopic.get(topic);
+            for (int partition = 0; partition < partitions; partition++) {
+                allPartitions.add(new TopicPartition(topic, partition));
+            }
+        }
+        return allPartitions;
+    }
+
+    @Override
+    public String name() {
+        return "round-robin";
+    }
+
+    @Override
+    public short version() {
+        return 0;
+    }
+
+    private static class CircularIterator<T> implements Iterator<T> {
```

Comment:
This doesn't currently build for me because Iterator's `remove()` method is not implemented.